crates/ruff_linter/src/rules/refurb/rules/fstring_number_format.rs

```diff
@@ -0,0 +1,185 @@
+use crate::fix::snippet::SourceCodeSnippet;
+
+/// ## What it does
+/// Checks for uses of `bin(...)[2:]` (or `hex`, or `oct`) to convert
+/// an integer into a string with no base prefix.
+///
+/// ## Why is this bad?
+/// Using f-strings is more concise and readable than using the
+/// `bin`, `hex`, or `oct` functions followed by slicing them.
+///
+/// ## Example
+/// ```python
+/// print(bin(1337)[2:])
+/// ```
+///
+/// Use instead:
+/// ```python
+/// print(f"{1337:b}")
+/// ```
+#[violation]
+pub struct FStringNumberFormat {
+    replacement: Option<SourceCodeSnippet>,
+    int_base: IntBase,
+}
+
+impl Violation for FStringNumberFormat {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let FStringNumberFormat {
+            replacement,
+            int_base,
+        } = self;
+        let function_name = int_base.function_name();
+
+        if let Some(snippet) = replacement {
+            if let Some(full_display) = snippet.full_display() {
+                return format!("Replace `{function_name}` parsing with `{full_display}`.");
+            }
+        }
+        format!("Replace `{function_name}` parsing with f-string.")
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        let FStringNumberFormat {
+            replacement,
+            int_base,
+        } = self;
+        let function_name = int_base.function_name();
+
+        if let Some(snippet) = replacement {
+            if let Some(full_display) = snippet.full_display() {
+                return Some(format!(
+                    "Replace `{function_name}` parsing with `{full_display}`."
+                ));
+            }
+        }
+        Some(format!("Replace `{function_name}` parsing with f-string."))
+    }
+}
+
+/// FURB116
+pub(crate) fn fstring_number_format(checker: &mut Checker, subscript: &ast::ExprSubscript) {
+    // Validate the slice to be what we expect, `[2:]`
+    let Expr::Slice(ast::ExprSlice {
+        lower,
+        upper,
+        step,
+        range: _,
+    }) = subscript.slice.as_ref()
+    else {
+        return;
+    };
+
+    match (lower, upper, step) {
+        (Some(lower), None, None) => {
+            let Expr::NumberLiteral(ast::ExprNumberLiteral { value, .. }) = lower.as_ref() else {
+                return;
+            };
+
+            // Ensure the starting slice value is 2
+            if value.as_int().is_some_and(|v| *v != 2) {
+                return;
+            }
+        }
+        _ => return,
+    }
+
+    // Validate the value to be an integer base-changing function call
+    let Expr::Call(ExprCall {
+        func,
+        arguments,
+        range: _,
+    }) = subscript.value.as_ref()
+    else {
+        return;
+    };
+
+    let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() else {
+        return;
+    };
+
+    if !checker.semantic().is_builtin(id) {
+        return;
+    }
```

Comment:
If #10919 is merged before this, I'll update accordingly.

Reply:
it was merged ;)